src/main/kotlin/no/nav/syfo/behandler/BehandlerService.kt

```diff
@@ -67,14 +66,18 @@ class BehandlerService(
             )
         }
 
-        val pBehandlereForArbeidstakerIdList =
-            database.getBehandlerForArbeidstaker(
+        val pBehandlereForArbeidstakerList =
+            database.getBehandlerForArbeidstakerMedType(
                 personIdentNumber = behandlerArbeidstakerRelasjon.arbeidstakerPersonident,
-            ).map { it.id }
+            )
+
+        val isBytteAvFastlege = behandlerArbeidstakerRelasjon.type == BehandlerArbeidstakerRelasjonType.FASTLEGE &&
+            pBehandlereForArbeidstakerList
+            .filter { it.second == BehandlerArbeidstakerRelasjonType.FASTLEGE.name }
+            .map { it.first.id }.firstOrNull() != pBehandler.id
 
-        val isBytteAvBehandler = pBehandlereForArbeidstakerIdList.firstOrNull() != pBehandler.id
-        val behandlerIkkeKnyttetTilArbeidstaker = !pBehandlereForArbeidstakerIdList.contains(pBehandler.id)
-        if (isBytteAvBehandler || behandlerIkkeKnyttetTilArbeidstaker) {
+        val behandlerIkkeKnyttetTilArbeidstaker = !pBehandlereForArbeidstakerList.map { it.first.id }.contains(pBehandler.id)
+        if (isBytteAvFastlege || behandlerIkkeKnyttetTilArbeidstaker) {
```

Comment:
Kan vurdere å bruke destrukturering i stedet for `it` i denne funksjonen for å gjøre det litt mer lesbart.
Altså type  `( _, behandler ) ->` og `( behandler, _ ) ->` is stedet for `it`

Reply:
Ja, forsøker det.

Reply:
Synes det ble lettere å lese nå 👍